profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -142,107 +195,201 @@ std::string variorum_print_power_call() {
-      }
-    } else if (type_of_profiling == 2) {  // if only json is selecte
-      if (mpiOutPut) {
-        std::string filenamejson = printpath + "variorum-output-mpi-rank-" +
-                                   std::to_string(rank) + "-json.txt";
-        std::ofstream file;
-        file.open(filenamejson, std::ios_base::app);
-        char* s = variorum_json_call();
-        file << s;
-      } else {
-        std::cout << "MPI Rank " << rank << "\n";
-        char* s = variorum_json_call();
-        puts(s);
-        std::cout << s << std::endl;
-      }
-    }
-    file.close();
-  }
-#endif
-}
-
 // Function: variorum_call
 // Description: The function determines what profiling options are selected and
 // prints the profoiling data out to std out Pre: None Post: An output message
 // if variourum returned an error or if it functioned correctly
-
 void variorum_call() {
-  std::string output;
+  // std::cout <<output  << std::endl;
+  long long temp = 0;
+
+  int num_sockets = 0;
+
   if (type_of_profiling == 0) {
     output  = variorum_print_power_call();
     char* s = variorum_json_call();
-    std::cout << s << "\n";
-    std::cout << output << std::endl;
-  } else if (type_of_profiling == 1) {
-    output = variorum_print_power_call();
-    std::cout << output << std::endl;
-  } else if (type_of_profiling == 2) {
-    char* s = variorum_json_call();
-    std::cout << s << std::endl;
+
+    power += 1;
+
+    if (power != 0) {
+      gpu_powers.clear();
+      gpu_powers2.clear();
+    }
+    // std::cout<< s << std::endl;
+
+    json_error_t error;
+    json_t* root            = NULL;
+    json_t* socket_0        = NULL;
+    json_t* timestamp_value = NULL;
+    json_t* power_gpu_watts = NULL;
+    json_t* gpu_0_value     = NULL;
+
+    // Parse JSON string into a json_t object
+    root = json_loads(s, 0, &error);
+    if (!root) {
+      fprintf(stderr, "Error parsing JSON: %s\n", error.text);
+    }
+
+    const char* key;
+    json_t* value;
+    json_object_foreach(root, key, value) {
+      if (json_is_object(value)) {
+        json_t* socket_object = json_object_get(value, "socket_0");
+        if (socket_object && json_is_object(socket_object)) {
+          socket_0        = socket_object;
+          timestamp_value = json_object_get(value, "timestamp");
+          if (time1 == 0) {
+            time1 = (long long)json_integer_value(timestamp_value);
+          } else {
+            time2 = (long long)json_integer_value(timestamp_value);
+          }
+          break;
+        }
+      }
+    }
+
+    if (!socket_0 || !timestamp_value) {
+      fprintf(stderr, "Failed to find 'socket_0' object or 'timestamp'.\n");
+    }
+
+    // Access power_gpu_watts within socket_0
+    power_gpu_watts = json_object_get(socket_0, "power_gpu_watts");
+    if (!json_is_object(power_gpu_watts)) {
+      fprintf(stderr, "Expected 'power_gpu_watts' to be an object.\n");
+    }
+    if (gdevID == 0) {
```

Comment:
In my testing I have been getting device 0 often and came to the conclusion that it will be the most often used therefore the code bases getting the GPU powers off the assumption GPU 0 is where the information the user needs is.

Reply:
The reason why you see GPU 0 exclusively is that you will only use a different GPU if running with MPI (and multiple GPUs are present) or if you are requesting a different GPU explicitly at initialization, see https://kokkos.org/kokkos-core-wiki/ProgrammingGuide/Initialization.html?highlight=device+id#initialization-by-command-line-arguments.